llvm/test/Transforms/ConstraintElimination/pr70510.ll

```diff
@@ -0,0 +1,94 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 3
+; RUN: opt -passes=constraint-elimination -S %s | FileCheck %s
+
+define i32 @main() {
```

Comment:
nit: might be good to make the test name more explicit, e.g. test_loop_varying_bound_add_rec